polyfill/lib/ecmascript.mjs

```diff
@@ -469,7 +477,7 @@ export function ParseTemporalInstantString(isoString) {
 
 export function ParseTemporalZonedDateTimeString(isoString) {
   const result = ParseISODateTime(isoString);
-  if (!result.ianaName) throw new RangeError('Temporal.ZonedDateTime requires a time zone ID in brackets');
+  if (!result.tzName) throw new RangeError('Temporal.ZonedDateTime requires a time zone ID in brackets');
```

Comment:
What part of [ParseTemporalZonedDateTimeString](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporalzoneddatetimestring) does this `throw` correspond with?

Reply:
I'm not super-familiar with this part of the polyfill, but from a quick look it seems that the spec validates each expected format against the required production before calling ParseISODateTime. The polyfill first calls ParseISODateTime, and then validates that the result has the required elements from the corresponding production.

From a quick review, this divergence seems OK as an alternative to having super-complicated regexes in the polyfill for each type. And it probably makes the requirements for each type a bit more clear than readers trying to puzzle out syntax requirements in regex-ese.

I don't see this as being important enough to change the polyfill (or the spec), but happy to hear your perspective.

To answer your question, AFAICT it's this step:

> 1. If [ParseText](https://tc39.es/ecma262/#sec-parsetext)([StringToCodePoints](https://tc39.es/ecma262/#sec-stringtocodepoints)(isoString), [TemporalZonedDateTimeString](https://tc39.es/proposal-temporal/#prod-TemporalZonedDateTimeString)) is a [List](https://tc39.es/ecma262/#sec-list-and-record-specification-type) of errors, throw a RangeError exception.

Note that validating the time zone ID annotation is just one of the possible errors. The rest of them are AFAICT thrown in ParseISODateTime itself.